src/main/scala/com/databricks/spark/sql/perf/Benchmark.scala

```diff
@@ -413,27 +419,47 @@ abstract class Benchmark(@transient protected val sqlContext: SQLContext)
         }
 
         val breakdownResults = if (includeBreakdown) {
-          val depth = queryExecution.executedPlan.treeString.split("\n").size
+          val depth = queryExecution.executedPlan.collect { case p: SparkPlan => p }.size
           val physicalOperators = (0 until depth).map(i => (i, queryExecution.executedPlan(i)))
-          physicalOperators.map {
+          val indexMap = physicalOperators.map { case (index, op) => (op, index) }.toMap
+          val timeMap = new mutable.HashMap[Int, Double]
+
+          physicalOperators.reverse.map {
             case (index, node) =>
+              messages += s"Breakdown: ${node.simpleString}"
+              val newNode = buildDataFrame.queryExecution.executedPlan(index)
               val executionTime = benchmarkMs {
-                node.execute().map(_.copy()).foreach(row => Unit)
+                node.execute().foreach((row: Any) => Unit)
               }
-              BreakdownResult(node.nodeName, node.simpleString, index, executionTime)
+              timeMap += ((index, executionTime))
+
+              val childIndexes = node.children.map(indexMap)
```

Comment:
Here I think we actually want to use the original in case attribute ids or anything change amongst trees.